specifyweb/context/views.py

```diff
@@ -570,14 +570,34 @@ def views(request):
 @login_maybe_required
 @cache_control(max_age=86400, private=True)
 def viewsets(request):
-    """Retrive a list of Specify 6 viewset xml files."""
-    viewsets = []
-    for root, dir, files in os.walk(settings.SPECIFY_CONFIG_DIR):
-        for file in files:
-            if file.endswith('.views.xml'):
-                viewsets.append(
-                    os.path.relpath(os.path.join(root, file),settings.SPECIFY_CONFIG_DIR)
-                )
+    """
+    Retrieve a list of Specify 6 viewset xml files.
+    Filter out viewsets that are not applicable to the user type.
+    Filter out viewsets that are in FORM_RESOURCE_EXCLUDED_LST.
+    """
+    # Get a list of the acceptable viewsets for the user type.
+    FORM_USER_HIERARCHY = ['Admin', 'Manager', 'FullAccess', 'LimitedAccess', 'Guest']
+    USER_TYPE_DIR_NAME_MAP = {'Admin': 'admin', 'Manager': 'manager', 'FullAccess': 'fullaccess', 
```

Comment:
hmm, I had no idea Admin was a user type
did you get that from here?

https://github.com/specify/specify7/blob/2f39cf0a9b023fe28e25c82b0e35e311567f2790/specifyweb/specify/ldap_extras.py#L15-L21

there are no views for Admin user type in sp6 at the moment, and don't think we are going to add some, so probably don't need to handle that case here

Reply:
I agree with @maxpatiiuk – we shouldn't need to handle it

Reply:
@grantfitzsimmons ya, looking through the file system, it seems just 'Manager' and 'Guest' are used.  Should just those be supported, or should 'FullAccess' and 'LimitedAccess' be looked for in the file path as well?